### ISSUE TITLE
vmalert: support multi-doc yaml parsing

### ISSUE THREAD

**Issue by hagen1778** [VictoriaMetrics/VictoriaMetrics#6753]
### Is your feature request related to a problem? Please describe

vmalert can be configured to read rules via `-rule` cmd-line flag. The content of the file/url in the `-rule` is supposed to contain one valid yaml document. However, yaml supports multi-documents - see https://gettaurus.org/docs/YAMLTutorial/#YAML-Multi-Documents

### Describe the solution you'd like

Would be great if vmalert could try its best during config parsing and check if file/url contains a multi-doc. The naive approach will be to do `strings.Split(doc, "\n---\n")` before parsing the file.

### Describe alternatives you've considered

Do not use multi docs.

**Comment by Irene-123** [comment]
Hi @hagen1778 
I went through the issue and looking to contribute here, though need some time for more clarification and understanding
Wanted to know if I can take up this as my first issue here or if you have any suggestions lmk :)
Thanks!

**Comment by hagen1778** [comment]
Hi @Irene-123! 
This looks like a good first issue. 
The problem this enhancement should solve is that users try concatenating multiple config files into one.
For example:
```
groups:
- name: group1
  rules:
  - alert: ServiceDown
    expr: up == 0
    labels:
      severity: critical
---
groups:
- name: group2
  rules:
  - alert: TooManyErrors
    expr: rate(http_errors_total[5m]) > 0
    for: 1m
    labels:
      severity: critical
```
This config should be parsed by vmalert just like it get two config files instead of one. Probably, in [this function](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/a99fcfbf1a3df8243af039626d0ae0d0b5c5a2ab/app/vmalert/config/config.go#L296-L311) we can add an extra logic to split the content by `"\n---\n"` marker before parsing. And then return merged list of groups in response.

**Comment by Irene-123** [comment]
Yes, thank you for providing the information! 
I'll start working on it, and let you know :)

**Comment by f41gh7** [comment]
The feature was included into [v1.105.0](https://docs.victoriametrics.com/changelog/#v11050)

 Thanks @Irene-123  for the implementation
